src/Functions/FunctionsStringSearch.h

```diff
@@ -174,12 +193,28 @@ class FunctionsStringSearch : public IFunction
                 column_start_pos,
                 vec_res);
         else if (col_haystack_vector && col_needle_const)
-            Impl::vectorConstant(
-                col_haystack_vector->getChars(),
-                col_haystack_vector->getOffsets(),
-                col_needle_const->getValue<String>(),
-                column_start_pos,
-                vec_res);
+        {
+            if constexpr (execution_error_policy == ExecutionErrorPolicy::Null)
```

Comment:
What if we are not in mode "haystack-is-a-vector-and-needle-is-a-constant: but in l. 219, 227 or 233? I suppose similar modifications need to be done for these cases too.

Reply:
In case of `hasToken` and `hasTokenCaseInsensitive` only `vectorConstant` is implemented, so I only wished to apply my modifications to this particular one. However, if you deem that this would be a worthy addition then I could add the extra parameters to **all** such functions; thus eliminating the need for differentiation using `if constexpr`. What do you think?

Reply:
In my view, it is okay to not add code for stuff which isn't used anyways. At the same time, we should make assumptions explicit. What about this:
- add a new `res_null` parameter to the other three methods (`vectorFixedVector`, `vectorFixedConstant` and `constantVector`). This parameter will be mandatory / non-optional and you can simply pass in `nullptr` from FunctionsStringSearch.h.
- Then add `assert(!res_null)` to the beginning of all implementations of `vectorFixedVector`, `vectorFixedConstant` and `constantVector` (i.e. as preconditions). This should make clear that the implementations are not supposed to read or write `res_null`.

Reply:
I agree completely, because I prefer generic solutions. Since we're on the same page I'm happy to put in the extra work.